packages/runtime/container-runtime/src/test/batching.spec.ts

```diff
@@ -138,106 +146,249 @@ describe("Runtime batching", () => {
+				"Batch end without batch start should fail",
+			);
+		});
+
+		it("fails processing a batch with multiple batch starts", async () => {
+			const messageCount = 5;
+			const batch = getMessages(messageCount);
+			batch[2].metadata = { batch: true };
+
+			assert.throws(
+				() => processBatch(batch, containerRuntime),
+				(e: Error) => validateAssertionError(e, "Unexpected batch start marker"),
+				"Batch with multiple batch starts should fail",
+			);
+		});
+
+		it("fails processing a batch containing ops from multiple clients", async () => {
+			const messageCount = 5;
+			const batch = getMessages(messageCount);
+			// Change the clientId of the second message to a different client.
+			batch[1].clientId = "otherClientId";
+
+			assert.throws(
+				() => processBatch(batch, containerRuntime),
+				(e: any) => {
+					assert(e.errorType === FluidErrorTypes.dataCorruptionError);
+					assert(e.message === "Received messages from multiple clients in a batch");
+					return true;
+				},
+				"Batch with ops from multiple clients should fail",
+			);
+		});
+
+		it("fails processing a batch containing a non-runtime op along with runtime ops", async () => {
+			const messageCount = 5;
+			const batch = getMessages(messageCount);
+			// Change the type of the second message to a non-runtime op.
+			batch[1].type = MessageType.NoOp;
+
+			assert.throws(
+				() => processBatch(batch, containerRuntime),
+				(e: any) => {
+					assert(e.errorType === FluidErrorTypes.dataProcessingError);
+					assert(e.message === "Received out-of-order messages in batch");
+					return true;
+				},
+				"Batch with non-runtime op along with runtime ops should fail",
+			);
+		});
+
+		it("fails processing a batch containing an unknown runtime op along with known ops", async () => {
+			const messageCount = 5;
+			const batch = getMessages(messageCount);
+
+			// Change the type of the second message to an unknown runtime op.
+			const unknownMessage = batch[1];
+			const unknownMessageType = "unknown";
+			(unknownMessage.contents as any).type = unknownMessageType;
+
+			assert.throws(
+				() => processBatch(batch, containerRuntime),
+				(e: any) => {
+					assert(e.errorType === FluidErrorTypes.dataProcessingError);
+					assert(e.message === "Runtime message of unknown type");
+					return true;
+				},
+				"Batch with unknown runtime op along with known ops should fail",
+			);
+		});
 	});
 
-	it("fails processing a batch containing a non-runtime op along with runtime ops", async () => {
-		const messageCount = 5;
-		const batch = getBatch(messageCount);
-		// Change the type of the second message to a non-runtime op.
-		batch[1].type = MessageType.NoOp;
-
-		assert.throws(
-			() => processBatch(batch, containerRuntime),
-			(e: any) => {
-				assert(e.errorType === FluidErrorTypes.dataProcessingError);
-				assert(e.message === "Received a system message during batch processing");
-				return true;
+	describe("Delta scheduler for batches", () => {
```

Comment:
Could you check in these tests in a separate PR like you did with the previous block?

Reply:
This needs the change I am making in this PR. So, I could do this as a follow up and skip the existing tests for now. IMO, having it part of this PR makes sense but if it helps with review, I could do a follow up.

Reply:
Nah it's fine, it's just always reassurring the have the tests there and not changing (and still passing) when making a refactor like this.